schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/SchemaMetadataStorable.java

```diff
@@ -224,8 +230,17 @@ public static SchemaMetadataStorable updateSchemaMetadata(SchemaMetadataStorable
         schemaMetadataStorable.setSchemaGroup(schemaMetadata.getSchemaGroup());
         schemaMetadataStorable.setName(schemaMetadata.getName());
         schemaMetadataStorable.setDescription(schemaMetadata.getDescription());
-        schemaMetadataStorable.setCompatibility(schemaMetadata.getCompatibility());
-        schemaMetadataStorable.setValidationLevel(schemaMetadata.getValidationLevel());
+
+        schemaMetadataStorable.setCompatibility(
```

Comment:
If this is the case would it not be better to keep still the logic in schema metadata and move to the getter methods to have the null check instead of having to null check many times in code. (Or needing to remember to do so)